Code/Framework/AzCore/Platform/Common/WinAPI/AzCore/IO/SystemFile_WinAPI.cpp

```diff
@@ -350,6 +350,12 @@ namespace Platform
         return GetAttributes(fileName) != INVALID_FILE_ATTRIBUTES;
     }
 
+    bool IsDirectory(const char* filePath)
+    {
+        DWORD attributes = GetAttributes(filePath);
+        return attributes != INVALID_FILE_ATTRIBUTES && (GetAttributes(filePath) & FILE_ATTRIBUTE_DIRECTORY) != 0;
```

Comment:
Since `GetAttributes` is already called on `filePath` it doesn't have to be called again in the return-statement.
Shouldn't this take `filePath`, convert it from utf8 to wide-characters and call `GetAttributesW`? The original implementation seems to do this as well.

Reply:
Actually `GetAttributes` is our function.
It already takes the filepath and converts the UTF-8 string to a wchar_t* string.

Reply:
Now that function isn't actually treating the string as UTF-8 however.
It treats it as a multibyte string that gets converted based on the current locale

Reply:
I have updated the functions to use our `AZStd::(w)string` function which are unicode aware.